xarray/backends/netcdf3.py

```diff
@@ -62,6 +62,13 @@ def coerce_nc3_dtype(arr):
     dtype = str(arr.dtype)
     if dtype in _nc3_dtype_coercions:
         new_dtype = _nc3_dtype_coercions[dtype]
+        # check if this looks like a time with NaT
+        # and transform to float64
+        if np.issubdtype(dtype, np.int64):
+            mask = arr == np.iinfo(np.int64).min
+            if mask.any():
+                arr = np.where(mask, np.nan, arr)
+                return arr
```

Comment:
@spencerkclark I agree, your argument with the non-time related int64 containing `np.iinfo(np.int64).min` is very valid. OTOH, we do not know how many legacy workflows exist out there which would break. 

In the light of this how about moving this check to the variable-level where we have the attributes available? I'll try to come up with something reasonable.

Reply:
I've moved the nc3 handling to `encode_nc3_variable`.